src/generic/matrices.h

```diff
@@ -359,6 +359,15 @@ namespace oomph
     virtual void multiply_transpose(const DoubleVector& x,
                                     DoubleVector& soln) const = 0;
 
+    // compute the infinity-norm
+    virtual double inf_norm() const
+    {
+      throw OomphLibError(
```

Comment:
Odd. Is this function implemented for any specific matrix class? Have you forgotten to commit the relevant code? If not there isn't much point committing these changes since nobody would be able to test the code (which I,m sure you have done, somewhere)

Reply:
currently it's only been implemented for CRDoubleMatrix, but I wanted to maintain generality by having compute_condition_number(DoubleMatrixBase* const&) instead of the specific compute_condition_number(CRDoubleMatrix* const&) so that in the future if someone needs this for say a CCDoubleMatrix the function doesn't need to change, they just need to write their own implementation of the inf-norm for a CC matrix.

Reply:
exactly - it already existed just for CRDoubleMatrix, so I've just added a virtual function to the base class so that CRDoubleMatrix::inf_norm() is now an override of this broken DoubleMatrixBase::inf_norm(). And I think this is conceptually correct, since all matrices have an infinity norm so it makes sense to have this functionality in the base class.

Agreed on the self-test.

Reply:
...and did you write one?